### PR TITLE
Add Warmup Operation in Perf-tool Benchmarking Tests

### PR DESCRIPTION
**Pull request by naveentatikonda**
### Description
Add Warmup Operation in Perf-tool Benchmarking Tests.
 
### Check List
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by naveentatikonda** [comment]
Build failure is not related to these changes.

**Comment by navneet1v** [comment]
@naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf

File: benchmarks/perf-tool/okpt/test/steps/steps.py [hunk 1]
```diff
@@ -690,6 +709,23 @@ def delete_model(endpoint, port, model_id):
...
+    Performs warmup operation on index to load native library files
+    of that index to reduce query latencies.
+    Args:
+        endpoint: Endpoint OpenSearch is running on
+        port: Port OpenSearch is running on
+        index: index name
+    Returns:
+        number of shards the plugin succeeded and failed to warm up.
+    """
+    response = requests.get('http://' + endpoint + ':' + str(port) +
```
Comment (martin-gaievski) [inline benchmarks/perf-tool/okpt/test/steps/steps.py:723]: Seems we're using default timeout here, what is the chance that request will take longer than that time?

**Comment by naveentatikonda** [comment]
> @naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf

I have already added it in all release config. I missed a couple of them in sample config. I have added it now.

File: benchmarks/perf-tool/okpt/test/steps/steps.py (hunk 1 above)
Comment (naveentatikonda) [reply benchmarks/perf-tool/okpt/test/steps/steps.py:723]: I think our warmup API doesn't accept a timeout parameter. So, I think we can add a static sleep time of 5 min. WDYT ?

Comment (martin-gaievski) [reply benchmarks/perf-tool/okpt/test/steps/steps.py:723]: that should work. We can do it as a parameter for this step, and use 5 mins as a default value

**Comment by navneet1v** [comment]
> > @naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf
> 
> I have already added it in all release config. I missed a couple of them in sample config. I have added it now.

You missed on the filtering configs for the algorithms

File: benchmarks/perf-tool/okpt/test/steps/steps.py (hunk 1 above)
Comment (naveentatikonda) [reply benchmarks/perf-tool/okpt/test/steps/steps.py:724]: As discussed offline, made the timeout parameter configurable to fix the issue

**Comment by naveentatikonda** [comment]
> > > @naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf
> > 
> > 
> > I have already added it in all release config. I missed a couple of them in sample config. I have added it now.
> 
> You missed on the filtering configs for the algorithms

Oh sorry, Thanks for catching it. Updated them.

**Review by martin-gaievski — APPROVED** [review]

**Review by navneet1v — APPROVED** [review]

**Pull request merged**